### DISCUSSION TITLE
School app?

### DISCUSSION THREAD

**Discussion by Fresinosamboko** [community/community#87484] [Pull Requests] [answered]
### Select Topic Area

General

### Body

How to create app? Help me.

**Comment by mzums** [comment]
1. First choose a framework. Personally, I recommend Flutter but you can also use React Native.
Before writing the app try to learn the basics of what you will need.

2. Second, create a design that you will follow.

3. Third, write and test the app.
On what systems should it run? (Flutter and React Native support various operation systems including Android and IOS)

4. If necessary make the server part (here I recommend Python). But I guess you won't need it for a school project.

**Reply by mishmanners** [reply, marked as answer]
There are also lots of places online to learn how to create and build things. For example [dev.to](https://dev.to) is a good place to find tutorials and learn. There's also the GitHub repo I put together for free learning resources: https://github.com/mishmanners/Free-Learning-Content

Make sure you're not just making something for the sake of making something. Have it to something/solve a problem. You can always start with something or some issue that you experience each day, or something you want to solve for yourself. Even if it's for a school project, pick something that will be useful.

Work out if you're creating a web app/mobile app. What do you want the app to do? What functionality do you want it to have? How complex does the app need to be.

Decide what programming language you are going to use and what framework. For a bit of fun, you can check this out: https://youtu.be/pgrGSnC3SKE

And finally, if you're a student, you can get lots of free resources such as hosting credits and more with the Student Developer Pack. Head over to [GitHub Global Campus](https://education.github.com/) and register with your student email address for free to get access to hundreds of free things, including GitHub Copilot.
